ggshield/utils/git_shell.py

```diff
@@ -265,6 +285,28 @@ def get_last_commit_sha_of_branch(branch_name: str) -> Optional[str]:
     return last_target_commit[0]
 
 
+def get_repository_url_from_path(wd: Path) -> Optional[str]:
+    """
+    Returns one of the repository remote urls. Returns None if no remote are found,
+    or the directory is not a repository.
+    """
+    try:
+        if not is_git_dir(wd):
+            return None
+        remotes_raw = git(["remote", "-v"], cwd=wd).splitlines()
+        matches = (re.search(r"^.*\t(.*) \(.*\)$", remote) for remote in remotes_raw)
```

Comment:
nit: performance may not be ideal in this regex. However, we shouldn't have too many/long items

Reply:
I agree, but in most cases there will only be two remote lines here (one for fetch, one for push), and many characters seem to be allowed in remote names